### PR TITLE
Add aggregation metrics

### PR DESCRIPTION
**Pull request by cbalioglu**
This PR subclasses the torcheval aggregation metrics and relaxes the type requirement for their `input` parameter from being Tensor-only to Tensor, int, or float.

### CODE REVIEW

File: src/fairseq2/recipes/wav2vec2/asr/common.py [hunk 1]
```diff
@@ -64,11 +64,9 @@ def update_ctc_loss(self, batch: Seq2SeqBatch, loss: Tensor) -> None:
         :param ctc_loss:
             The loss of ``batch``.
         """
-        batch_size = torch.tensor(batch.batch_size)
+        normalized_loss = loss / batch.batch_size / math.log(2)
 
-        normalized_loss = loss.cpu() / batch_size / math.log(2)
```
Comment (kauterry) [inline src/fairseq2/recipes/wav2vec2/asr/common.py:69]: Why was it `loss.cpu()` before?

Comment (cbalioglu) [reply src/fairseq2/recipes/wav2vec2/asr/common.py:69]: Just for a bit of consistency. `batch_size` was a CPU tensor, so wanted them to be on the same device.
